### PR TITLE
[Inductor] Make static_input_idxs a set for faster lookup

### PR DESCRIPTION
**Pull request by platers**
`static_input_idxs` is only used for lookups. With large models, this is a large list. This takes over a millisecond in some cases.

Profile before change:
image

Profile after change: gaps are smaller, 1ms speedup before launching the cuda graph
image

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by platers** [comment]
image
I also see similar behavior in torch dynamo, but wasn't able to find the source of this logic. Any pointers would be appreciated!

**Comment by ezyang** [comment]
I would stamp and merge this but we need to audit to ensure use of set doesn't introduce nondeterminism here

**Comment by ezyang** [comment]
@laithsakka This seems like a good microbenchmark candidate

File: torch/_inductor/compile_fx.py [hunk 1]
```diff
@@ -1037,7 +1037,7 @@ def cudagraphify_impl(
     Assumes inputs[static_input_idxs[i]] are always the same memory address
     """
     check_input_idxs = get_input_idxs_to_check(inputs, static_input_idxs)  # type: ignore[arg-type]
-    static_input_idxs = remove_unaligned_input_idxs(inputs, static_input_idxs)  # type: ignore[arg-type]
+    static_input_idxs = set(remove_unaligned_input_idxs(inputs, static_input_idxs))  # type: ignore[arg-type]
```
Comment (oulgen) [inline torch/_inductor/compile_fx.py:1040]: Please use `OrderedSet` for deterministic behavior

**Review by oulgen — CHANGES_REQUESTED** [review]

File: torch/_inductor/compile_fx.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/compile_fx.py:1040]: @oulgen if we have blanket banned regular set from inductor, even if it's used in a way that wouldn't cause nondeterminism, we should have a lint for it.

Comment (oulgen) [reply torch/_inductor/compile_fx.py:1040]: @eellison was writing a linter, not sure how that's going

**Review by oulgen — DISMISSED** [review]

**Comment by ZainRizvi** [comment]
@pytrochbot revert -c nosignal -m "Lint is failing on this file in trunk. See [GH job link](https://github.com/pytorch/pytorch/actions/runs/10777258770/job/29885960050) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/011cae9570fb3c44b7f6f0c8004c470579ed21da)"

Please try rebasing the change before remerging

**Comment by ZainRizvi** [comment]
@pytorchbot  revert -c nosignal -m "Lint is failing on this file in trunk. See [GH job link](https://github.com/pytorch/pytorch/actions/runs/10777258770/job/29885960050) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/011cae9570fb3c44b7f6f0c8004c470579ed21da)"

Please try rebasing the change before remerging

**Comment by platers** [comment]
Should have fixed the lint issue. @oulgen could you rebase and merge?

**Review by oulgen — APPROVED** [review]

**Pull request closed without merging**